### DISCUSSION TITLE
Documentation (Regardless of how simple) on how to use ConfuserEx.MSBuild

### DISCUSSION THREAD

**Discussion by Smurf-IV** [mkaring/ConfuserEx#361] [Ideas]
`ConfuserEx.MSBuild` in
- Solution builds within Visual studio for release only
- Dev-Ops `yml` files
- installation into project (Yes I know Nuget - but still)
- Other gotcha's on creation of output dir's etc.

**Comment by mkaring** [comment]
Like [the one in the wiki](https://github.com/mkaring/ConfuserEx/wiki/MSBuild-Integration)?

**Reply by Smurf-IV** [reply]
Yes, That would be good linked to it from the Nuget Package and from the Home page.

**Reply by hhgm** [reply]
I have tried to include ConfuserEx in my dotnet core 2.2 project, and I have followed the above documentation... but I end up with a ton of errors when I build my debug (it will eventually go to a DevOps server to build there for release)

**My project file**

<project outputDir=".\Confused" baseDir=".\" xmlns="http://confuser.codeplex.com">
  <rule pattern="true" preset="normal" inherit="false">
    <protection id="anti debug" />
    <protection id="anti tamper" />
    <protection id="constants" />
    <protection id="watermark" />
  </rule>
</project>

**Just including the top returns to give an indication on what to do..**

Severity	Code	Description	Project	File	Line	Suppression State
Error		Failed to resolve a member, check if all dependencies are present in the correct version.	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		Could not resolve method/field: System.Collections.Generic.List`1<TMServerAPIContracts.Models.Tournament.TournamentClass> TMServerAPIContracts.Request.Tournament.ClassesSaveRequest::get_Classes() (TMServerAPIContracts, Version=3.1.1300.0, Culture=neutral, PublicKeyToken=null)	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		---BEGIN DEBUG INFO---	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		Installed Framework Versions:	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		    v2.0.50727  2.0.50727.4927	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		    v3.0  3.0.30729.4926	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		    v3.5  3.5.30729.4926	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		    v4	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		    Client  4.8.04084	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		    Full  4.8.04084	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		    v4.0	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		    Client  4.0.0.0	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		    v4.5 528372	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error			TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		Cached assemblies:	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		    TMServerAPI, Version=3.1.0.0, Culture=neutral, PublicKeyToken=null (C:\Users\hhgm\source\Workspaces\TM nextGen Dev\TMServerAPI\obj\x64\Debug\netcoreapp2.2\TMServerAPI.dll)	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        System.Runtime, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.Extensions.Configuration.UserSecrets, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        System.Diagnostics.Debug, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        System.Linq.Expressions, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        System.Reflection.Emit, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        System.Collections, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        System.Threading, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        System.Reflection.Emit.ILGeneration, Version=4.0.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.EntityFrameworkCore, Version=2.2.6.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        TMServerDb, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        TMServerAPIContracts, Version=3.1.1300.0, Culture=neutral, PublicKeyToken=null	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.EntityFrameworkCore.Relational, Version=2.2.6.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        RestSharp, Version=106.3.1.0, Culture=neutral, PublicKeyToken=598062e77f915f75	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        System.Runtime.Extensions, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        NLog, Version=4.0.0.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.Extensions.Configuration.Abstractions, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.AspNetCore.Hosting.Abstractions, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.AspNetCore.Server.Kestrel.Core, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.Extensions.Logging, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.Extensions.DependencyInjection.Abstractions, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.Extensions.DependencyInjection, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        AutoMapper, Version=9.0.0.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.AspNetCore.Http.Abstractions, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.Extensions.Logging.Abstractions, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Microsoft.Extensions.Options, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69	
Error		        Hangfire.Core, Version=1.7.12.0, Culture=neutral, PublicKeyToken=null	TMServerAPI	C:\Users\hhgm\.nuget\packages\confuser.msbuild\1.5.0\build\Confuser.MSBuild.targets	69
